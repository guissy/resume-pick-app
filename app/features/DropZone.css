.mainBox {
  flex: 1;
  min-height: 90px;
  margin-top: 20px;
  border: 1px dashed #ddd;
  cursor: pointer;
}
.dropBox {
  width: 100%;
  height: 100%;
  padding: 30px;
  display: flex;
  justify-content: center;
  align-content: center;
  box-sizing: border-box;
}

.placeholder {
  width: 100%;
  color: #eee;
}

.textCenter {
  text-align: center;
}

.upload {
  color: #666;
}

.mainBox:hover .upload {
  color: #aaa;
}
