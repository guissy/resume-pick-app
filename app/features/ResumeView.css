.wrap {
    color: #eeeeee;
    background: #333333;
}

.article {
    height: calc(100vh - 120px);
    overflow: auto;
    padding: 20px 20px 100px 20px;
}
mark {
    border-radius: 4px;
    padding: 0 3px;
    display: inline-block;
    color: #ffffff;
    background-color: #999999;
}

:global(.kw0) {
    background: #e10000;
}
:global(.kw1) {
    background: #e1e100;
}
:global(.kw2) {
    background: #00e100;
}
:global(.kw0.pos_2_0),
:global(.kw0.pos_1_9),
:global(.kw0.pos_1_8),
:global(.kw0.pos_1_7),
:global(.kw0.pos_1_6) {
    background: rgba(225, 0, 0, 0.8);
}
:global(.kw0.pos_1_5),
:global(.kw0.pos_1_4),
:global(.kw0.pos_1_3),
:global(.kw0.pos_1_2),
:global(.kw0.pos_1_1) {
    background: rgba(195, 0, 0, 0.6);
}
:global(.kw0.pos_1_0),
:global(.kw0.pos_0_9),
:global(.kw0.pos_0_8),
:global(.kw0.pos_0_7),
:global(.kw0.pos_0_6) {
    background: rgba(165, 0, 0, 0.4);
}
:global(.kw0.pos_0_5),
:global(.kw0.pos_0_4),
:global(.kw0.pos_0_3),
:global(.kw0.pos_0_2),
:global(.kw0.pos_0_1) {
    background: rgba(135, 0, 0, 0.2);
}
:global(.kw1.pos_2_0),
:global(.kw1.pos_1_9),
:global(.kw1.pos_1_8),
:global(.kw1.pos_1_7),
:global(.kw1.pos_1_6) {
    background: rgba(225, 225, 0, 0.8);
}
:global(.kw1.pos_1_5),
:global(.kw1.pos_1_4),
:global(.kw1.pos_1_3),
:global(.kw1.pos_1_2),
:global(.kw1.pos_1_1) {
    background: rgba(195, 195, 0, 0.6);
}
:global(.kw1.pos_1_0),
:global(.kw1.pos_0_9),
:global(.kw1.pos_0_8),
:global(.kw1.pos_0_7),
:global(.kw1.pos_0_6) {
    background: rgba(165, 165, 0, 0.4);
}
:global(.kw1.pos_0_5),
:global(.kw1.pos_0_4),
:global(.kw1.pos_0_3),
:global(.kw1.pos_0_2),
:global(.kw1.pos_0_1) {
    background: rgba(135, 135, 0, 0.2);
}
:global(.kw0.pos_2_0),
:global(.kw0.pos_1_9),
:global(.kw0.pos_1_8),
:global(.kw0.pos_1_7),
:global(.kw0.pos_1_6) {
    background: rgba(0, 225, 0, 0.8);
}
:global(.kw2.pos_1_5),
:global(.kw2.pos_1_4),
:global(.kw2.pos_1_3),
:global(.kw2.pos_1_2),
:global(.kw2.pos_1_1) {
    background: rgba(0, 195, 0, 0.6);
}
:global(.kw2.pos_1_0),
:global(.kw2.pos_0_9),
:global(.kw2.pos_0_8),
:global(.kw2.pos_0_7),
:global(.kw2.pos_0_6) {
    background: rgba(0, 165, 0, 0.4);
}
:global(.kw2.pos_0_5),
:global(.kw2.pos_0_4),
:global(.kw2.pos_0_3),
:global(.kw2.pos_0_2),
:global(.kw2.pos_0_1) {
    background: rgba(0, 135, 0, 0.2);
}
:global(.kw2.neg_1_5),
:global(.kw2.neg_1_4),
:global(.kw2.neg_1_3),
:global(.kw2.neg_1_2),
:global(.kw2.neg_1_1) {
    background: rgba(160, 160, 160, 0.8);
}
:global(.kw2.neg_1_0),
:global(.kw2.neg_0_9),
:global(.kw2.neg_0_8),
:global(.kw2.neg_0_7),
:global(.kw2.neg_0_6) {
    background: rgba(160, 160, 160, 0.4);
}
:global(.kw2.neg_0_5),
:global(.kw2.neg_0_4),
:global(.kw2.neg_0_3),
:global(.kw2.neg_0_2),
:global(.kw2.neg_0_1) {
    background: rgba(160, 160, 160, 0.2);
}
