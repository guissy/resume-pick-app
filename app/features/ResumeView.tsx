import React from 'react';

type Props = {
  resume: string;
  onClose: () => void;
};
const ResumeView: React.FC<Props> = ({ resume, onClose }) => {
  return (
    <div
      style={{
        color: '#ffffff',
        background: '#333333',
      }}
    >
      <header style={{ textAlign: 'right' }}>
        <button type="button" onClick={onClose}>
          <i className="fa fa-times fa-2x" />
        </button>
      </header>
      <article
        style={{
          height: 'calc(100vh - 120px)',
          overflow: 'auto',
          paddingBottom: 100,
          padding: 20,
        }}
      >
        {resume}
      </article>
    </div>
  );
};

export default ResumeView;
