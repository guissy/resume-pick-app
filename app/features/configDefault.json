[
  {
    "name": "核心项",
    "children": [
      {
        "name": "RN",
        "alias": ["react native", "rn", "weex"],
        "score": 1.5
      },
      {
        "name": "flutter",
        "alias": ["flutter", "dart"],
        "score": 1.5
      },
      {
        "name": "h5",
        "alias": [
          "webview",
          "uni-app",
          "uniapp",
          "umi-app",
          "ionic",
          "apicloud",
          "mobile",
          "hybird"
        ],
        "score": 1.5
      },
      {
        "name": "react",
        "alias": ["redux", "antd", "ant design"],
        "score": 1.5
      },
      {
        "name": "vue",
        "alias": ["vue2", "vuejs", "vue2", "element", "vant"],
        "score": 1.5
      },
      {
        "name": "vue3",
        "score": 1.5
      },
      {
        "name": "angular",
        "alias": [
          "ng",
          "ng2",
          "angular2",
          "angular4",
          "angular5",
          "angular6",
          "angular7",
          "angular8"
        ],
        "score": 1
      },
      {
        "name": "wxapp",
        "alias": ["微信", "小程序", "taro", "mpvue", "mpvue"],
        "score": 1.5
      },
      {
        "name": "node",
        "alias": ["nodejs", "express", "keo", "egg", "next", "typeorm", "pm2"],
        "score": 1.5
      },
      {
        "name": "electron",
        "alias": ["nwjs"],
        "score": 1.5
      },
      {
        "name": "state mgr",
        "alias": [
          "redux",
          "saga",
          "mobx",
          "vuex",
          "vuetify",
          "dva",
          "事件总线",
          "eventBus",
          "event bus"
        ],
        "score": 1
      },
      {
        "name": "es6",
        "alias": [
          "es5",
          "es6+",
          "es2015",
          "es2016",
          "es2017",
          "es7",
          "es8",
          "es9",
          "ecmascript"
        ],
        "score": 0.5
      },
      {
        "name": "w3c",
        "score": 0.5,
        "alias": ["mdn"]
      },
      {
        "name": "css3",
        "alias": [
          "flex",
          "grid",
          "css3",
          "rem",
          "less",
          "sass",
          "css-modules",
          "css in js",
          "css-in-js",
          "styled-components",
          "jss",
          "emotion"
        ],
        "score": 0.5
      }
    ]
  },
  {
    "name": "进阶",
    "children": [
      {
        "name": "typescript",
        "alias": ["ts"],
        "score": 1.5
      },
      {
        "name": "rxjs",
        "alias": ["ReactiveX", "ngrx"],
        "score": 1.5
      },
      {
        "name": "graphql",
        "alias": ["gql"],
        "score": 1.5
      },
      {
        "name": "test",
        "calc": "max",
        "alias": [
          "mocha",
          "jasmine",
          "jest",
          "cypress",
          "puppeteer",
          "unit test"
        ],
        "score": 1.5
      },
      {
        "name": "lint",
        "alias": ["tslint", "eslint", "jslint", "命名规范", "代码规范"],
        "score": 0.5
      },
      {
        "name": "review",
        "alias": ["pr", "git flow", "代码审查"],
        "score": 0.5
      },
      {
        "name": "gayhub",
        "alias": ["github.com", "gitee.com"],
        "score": 1.5
      },
      {
        "name": "blog",
        "alias": [
          "个人博客",
          "博客地址",
          "csdn.net",
          "cnblogs.com",
          "juejin.im",
          "zhihu.com",
          "jianshu.com",
          "segmentfault.com",
          "oschina.net",
          "imooc.com",
          "51cto.com",
          "gitbooks.io"
        ],
        "score": 1.5
      }
    ]
  },
  {
    "name": "其他",
    "children": [
      {
        "name": "leader",
        "alias": ["组长", "负责人"],
        "score": 1
      },
      {
        "name": "mock",
        "score": 0.5
      },
      {
        "name": "git",
        "score": 0.5
      },
      {
        "name": "docker",
        "score": 0.5
      },
      {
        "name": "websocket",
        "alias": ["socket", "canvas", "charles"],
        "score": 0.5
      },
      {
        "name": "微服务",
        "score": 0.5
      },
      {
        "name": "加密",
        "alias": ["密文", "支付", "md5", "sha1", "签名", "白名单"],
        "score": 0.5
      },
      {
        "name": "重构",
        "alias": ["防抖", "去抖", "节流", "虚拟化", "懒加载"],
        "score": 0.5
      },
      {
        "name": "自定义",
        "alias": ["封装", "公共组件", "二次封装"],
        "score": 0.5
      },
      {
        "name": "python",
        "score": 0.5
      },
      {
        "name": "java",
        "score": 0.5
      },
      {
        "name": "go",
        "score": 0.5
      },
      {
        "name": "rust",
        "score": 0.5
      },
      {
        "name": "cpp",
        "score": 0.5
      },
      {
        "name": "swift",
        "score": 0.5
      },
      {
        "name": "kotlin",
        "score": 0.5
      },
      {
        "name": "scala",
        "score": 0.5
      },
      {
        "name": "mysql",
        "score": 0.5
      },
      {
        "name": "mongodb",
        "score": 0.5
      },
      {
        "name": "electron",
        "score": 0.5
      },
      {
        "name": "devops",
        "alias": ["shell", "nginx", "linux", "cicd"],
        "score": 0.5
      },
      {
        "name": "vim",
        "score": 0.5
      },
      {
        "name": "ssr",
        "score": 0.5
      },
      {
        "name": "bi",
        "score": 0.5,
        "alias": ["可视化", "大屏", "echarts", "d3"]
      },
      {
        "name": "three.js",
        "score": 0.5
      },
      {
        "name": "postman",
        "score": 0.5,
        "alias": ["charles", "fiddler"]
      },
      {
        "name": "leetcode",
        "score": 0.5,
        "alias": ["力扣", "leet code"]
      },
      {
        "name": "deno",
        "score": 0.5
      },
      {
        "name": "wasm",
        "score": 0.5,
        "alias": ["web assembly"]
      },
      {
        "name": "iframe",
        "alias": ["flash", "dreamweaver", "div+css", "v-for", "v-if"],
        "score": -1.5
      },
      {
        "name": "ajax",
        "alias": ["ajax", "jsonp", "ie6", "ie7", "ie8"],
        "score": -1
      },
      {
        "name": "jquery",
        "alias": ["jq", "zepto"],
        "score": -0.5
      },
      {
        "name": "svn",
        "alias": ["requirejs", "gulp", "grunt", "svn"],
        "score": -1
      },
      {
        "name": "积极乐观",
        "alias": ["为人和蔼", "吃苦耐劳", "上进", "了解"],
        "score": -0.5
      },
      {
        "name": "熟悉",
        "score": -0.3
      }
    ]
  }
]
