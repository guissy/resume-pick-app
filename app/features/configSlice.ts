import { createSlice } from '@reduxjs/toolkit';
// eslint-disable-next-line import/no-cycle
import { AppThunk, RootState } from '../store';
import { ConfigFile } from './type';
import initFile, { initDefaultFile, saveFile } from './configUtil';

export const configSlice = createSlice({
  name: 'config',
  initialState: { configs: {} as ConfigFile },
  reducers: {
    buildConfig: (state, action) => {
      if (Object.keys(state.configs).length === 0) {
        state.configs = action.payload;
      }
    },
    updateConfig: (state, action) => {
      Object.assign(state.configs, action.payload);
    },
    saveConfig: (state) => {
      saveFile(state.configs);
    },
    resetConfig: (state, action) => {
      state.configs = action.payload;
      saveFile(state.configs);
    },
  },
});

export const {
  updateConfig,
  buildConfig,
  saveConfig,
  resetConfig,
} = configSlice.actions;

export const initConfigAsync = (): AppThunk => (dispatch) =>
  initFile()
    .then((data) => dispatch(buildConfig(data)))
    // eslint-disable-next-line no-console
    .catch((e) => console.error(e));

export const resetConfigAsync = (): AppThunk => (dispatch) =>
  initDefaultFile()
    .then((data) => dispatch(resetConfig(data)))
    // eslint-disable-next-line no-console
    .catch((e) => console.error(e));

export default configSlice.reducer;

export const selectConfig = (state: RootState) => state.config.configs.default;
