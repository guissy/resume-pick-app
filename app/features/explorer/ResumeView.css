.wrap {
    color: #eeeeee;
    background: #333333;
}

.article {
    height: calc(100vh - 120px);
    overflow: auto;
    padding: 20px 20px 100px 20px;
}
mark {
    border-radius: 4px;
    padding: 0 3px;
    display: inline-block;
    color: #ffffff;
    background-color: #999999;
}

:global(.kw0) {
    background: #e10000;
}
:global(.kw1) {
    background: #e1e100;
}
:global(.kw2) {
    background: #00e100;
}
:global(.kws) {
    background: #ff00ff;
}
:global(.kw0.pos_2_0),
:global(.kw0.pos_1_9),
:global(.kw0.pos_1_8),
:global(.kw0.pos_1_7),
:global(.kw0.pos_1_6) {
    background: rgba(225, 0, 0, 0.8);
}
:global(.kw0.pos_1_5),
:global(.kw0.pos_1_4),
:global(.kw0.pos_1_3),
:global(.kw0.pos_1_2),
:global(.kw0.pos_1_1) {
    background: rgba(195, 0, 0, 0.6);
}
:global(.kw0.pos_1_0),
:global(.kw0.pos_0_9),
:global(.kw0.pos_0_8),
:global(.kw0.pos_0_7),
:global(.kw0.pos_0_6) {
    background: rgba(165, 0, 0, 0.4);
}
:global(.kw0.pos_0_5),
:global(.kw0.pos_0_4),
:global(.kw0.pos_0_3),
:global(.kw0.pos_0_2),
:global(.kw0.pos_0_1) {
    background: rgba(135, 0, 0, 0.2);
}
:global(.kw1.pos_2_0),
:global(.kw1.pos_1_9),
:global(.kw1.pos_1_8),
:global(.kw1.pos_1_7),
:global(.kw1.pos_1_6) {
    background: rgba(225, 225, 0, 0.8);
}
:global(.kw1.pos_1_5),
:global(.kw1.pos_1_4),
:global(.kw1.pos_1_3),
:global(.kw1.pos_1_2),
:global(.kw1.pos_1_1) {
    background: rgba(195, 195, 0, 0.6);
}
:global(.kw1.pos_1_0),
:global(.kw1.pos_0_9),
:global(.kw1.pos_0_8),
:global(.kw1.pos_0_7),
:global(.kw1.pos_0_6) {
    background: rgba(165, 165, 0, 0.4);
}
:global(.kw1.pos_0_5),
:global(.kw1.pos_0_4),
:global(.kw1.pos_0_3),
:global(.kw1.pos_0_2),
:global(.kw1.pos_0_1) {
    background: rgba(135, 135, 0, 0.2);
}
:global(.kw0.pos_2_0),
:global(.kw0.pos_1_9),
:global(.kw0.pos_1_8),
:global(.kw0.pos_1_7),
:global(.kw0.pos_1_6) {
    background: rgba(0, 225, 0, 0.8);
}
:global(.kw2.pos_1_5),
:global(.kw2.pos_1_4),
:global(.kw2.pos_1_3),
:global(.kw2.pos_1_2),
:global(.kw2.pos_1_1) {
    background: rgba(0, 195, 0, 0.6);
}
:global(.kw2.pos_1_0),
:global(.kw2.pos_0_9),
:global(.kw2.pos_0_8),
:global(.kw2.pos_0_7),
:global(.kw2.pos_0_6) {
    background: rgba(0, 165, 0, 0.4);
}
:global(.kw2.pos_0_5),
:global(.kw2.pos_0_4),
:global(.kw2.pos_0_3),
:global(.kw2.pos_0_2),
:global(.kw2.pos_0_1) {
    background: rgba(0, 135, 0, 0.2);
}
:global(.kw2.neg_1_5),
:global(.kw2.neg_1_4),
:global(.kw2.neg_1_3),
:global(.kw2.neg_1_2),
:global(.kw2.neg_1_1) {
    background: rgba(160, 160, 160, 0.8);
}
:global(.kw2.neg_1_0),
:global(.kw2.neg_0_9),
:global(.kw2.neg_0_8),
:global(.kw2.neg_0_7),
:global(.kw2.neg_0_6) {
    background: rgba(160, 160, 160, 0.4);
}
:global(.kw2.neg_0_5),
:global(.kw2.neg_0_4),
:global(.kw2.neg_0_3),
:global(.kw2.neg_0_2),
:global(.kw2.neg_0_1) {
    background: rgba(160, 160, 160, 0.2);
}

.searchWrap {
    display: inline-flex;
    align-items: center;
    margin-left: 20px;
}
.search {
    text-indent: 20px;
    color: #fff;
    width: 200px;
    font-size: 16px;
}
.searchWrap > i {
    transform: translateX(200%) scale(0.8);
}
.count {
    display: inline-block;
    font-size: 8px;
    color: #999999;
    border: 1px solid #545454;
    border-radius: 50%;
    width: 14px;
    height: 14px;
    line-height: 14px;
    text-align: center;
    transform: translateX(-50%);
}
