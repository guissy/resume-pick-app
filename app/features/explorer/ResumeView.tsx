import React, { useEffect } from 'react';
import { shell } from 'electron';
import { useKey } from 'react-use';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import Highlighter from 'react-highlight-words';
import { ScoreFile } from '../type';
import styles from './ResumeView.css';

type Props = {
  resume: ScoreFile | undefined;
  onClose: () => void;
};
const ResumeView: React.FC<Props> = ({ resume, onClose }) => {
  const topElmRef = React.useRef<HTMLParagraphElement>(null);
  useEffect(() => {
    if (resume && topElmRef.current) {
      topElmRef.current.scrollIntoView({ behavior: 'smooth' });
    }
  }, [resume]);
  const clickOpenNative = React.useCallback(() => {
    if (resume?.path) {
      if (resume?.path.startsWith('http')) {
        shell.openExternal(resume?.path);
      } else {
        shell.openItem(resume.path);
      }
    }
  }, [resume]);
  const [searchTxt, setSearchTxt] = React.useState('');
  const keywords = React.useMemo(
    () =>
      (resume?.keywords || [])
        .map((k) => k.children.map((w) => [w.name, ...(w.alias || [])]))
        .flat(2)
        .concat(searchTxt.split(' ')),
    [resume, searchTxt]
  );
  const count = React.useMemo(() => {
    const searchs = searchTxt?.split(' ').filter(Boolean);
    return searchTxt.trim()
      ? searchs?.reduce((s, v) => {
          const matches = resume?.text.match(new RegExp(v, 'gi')) || [];
          return s + matches.length;
        }, 0)
      : 0;
  }, [searchTxt, resume]);
  const keywordClassName = React.useMemo(
    () =>
      (resume?.keywords || [])
        .map((k, i) =>
          k.children.map((w) =>
            [w.name, ...(w.alias || [])].map((n) => ({
              [n]: `kw${i} ${w.score > 0 ? 'pos' : 'neg'}_${(w.gained / w.score)
                .toFixed(1)
                .replace('.', '_')}`,
            }))
          )
        )
        .flat(2)
        .concat(searchTxt.split(' ').map((s) => ({ [s]: 'kws' })))
        .reduce((s, v) => ({ ...s, ...v }), {}),
    [resume, searchTxt]
  );
  useKey('Escape', onClose);

  return (
    <div className={styles.wrap}>
      <header style={{ textAlign: 'right' }}>
        <label htmlFor="search" className={styles.searchWrap}>
          <i className="fa fa-search" />
          <span
            className={styles.count}
            style={{ visibility: count > 0 ? 'visible' : 'hidden' }}
          >
            {count}
          </span>
          <input
            className={styles.search}
            id="search"
            type="search"
            value={searchTxt}
            placeholder="输入关键字搜索简历"
            onChange={(e) => {
              setSearchTxt(e.target.value);
            }}
          />
        </label>
        <button
          type="button"
          onClick={clickOpenNative}
          style={{
            background: 'none',
            border: '1px solid #fff',
            color: '#fff',
            fontSize: 14,
            cursor: 'pointer',
          }}
        >
          使用本地软件打开此文件
        </button>
        <button type="button" onClick={onClose}>
          <i className="fa fa-times fa-2x" />
        </button>
      </header>
      <article className={styles.article} key={searchTxt}>
        {(resume?.text || '').split('\n').map((content, i) => (
          <p
            key={String(i)}
            style={{ marginBlockStart: 5, marginBlockEnd: 5 }}
            ref={i === 0 ? topElmRef : undefined}
          >
            <Highlighter
              highlightClassName={keywordClassName}
              searchWords={keywords}
              textToHighlight={content}
            />
          </p>
        ))}
      </article>
    </div>
  );
};

export default ResumeView;
