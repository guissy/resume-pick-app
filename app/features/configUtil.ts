// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import storage from 'electron-storage';
import configDefault from './configDefault.json';
import { Config } from './type';

function initFile(): Promise<Record<string, Config>> {
  return storage.get('keywords.json').catch((err: unknown) => {
    // eslint-disable-next-line no-console
    console.error(err);
    return storage.set('keywords.json', { default: configDefault });
  });
}

export function initDefaultFile(): Promise<Record<string, Config>> {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  return Promise.resolve({ default: configDefault });
}

export function saveFile(configs: Record<string, Config>) {
  return storage
    .set('keywords.json', JSON.stringify(configs, null, '  '))
    .catch((err: unknown) => {
      // eslint-disable-next-line no-console
      console.error(err);
    });
}

export default initFile;
