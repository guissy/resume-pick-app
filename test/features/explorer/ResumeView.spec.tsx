import '../utils/mockElectron';
import React from 'react';
import Enzyme, { mount, ReactWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { shell } from 'electron';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import Highlighter from 'react-highlight-words';
import mockScoreFile from '../utils/mockScoreFile';
import ResumeView from '../../../app/features/explorer/ResumeView';
import { ScoreFile } from '../../../app/features/type';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

const defaultTestProps = {
  resume: (mockScoreFile[0] as unknown) as ScoreFile,
  onClose: () => {},
};

function setup(props = defaultTestProps) {
  // eslint-disable-next-line react/jsx-props-no-spreading
  const getWrapper = () => mount(<ResumeView {...props} />);
  const component = getWrapper();
  return {
    component,
    highlighter: component.find('Highlighter').at(0) as ReactWrapper<
      Highlighter
    >,
    button: component.find('button').at(0),
    searchInput: component.find('input'),
  };
}
describe('ResumeView component', () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should match exact snapshot', () => {
    const tree = renderer
      // eslint-disable-next-line react/jsx-props-no-spreading
      .create(<ResumeView {...defaultTestProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show mark', () => {
    const { highlighter } = setup();
    expect(highlighter.exists()).toBeTruthy();
    expect(highlighter.find('mark').exists()).toBeTruthy();
  });

  it('click button should open file', () => {
    const { button } = setup();
    defaultTestProps.resume.path = '1.pdf';
    button.simulate('click');
    expect(shell.openItem).toBeCalledWith(defaultTestProps.resume.path);
    defaultTestProps.resume.path = 'https://baidu.com';
    button.simulate('click');
    expect(shell.openExternal).toBeCalledWith(defaultTestProps.resume.path);
  });

  it('input should search', async () => {
    const { getByLabelText, getByText } = render(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <ResumeView {...defaultTestProps} />
    );
    const searchInput = getByLabelText(/search input/i);
    fireEvent.change(searchInput, { target: { value: 'good' } });
    expect(searchInput.getAttribute('value')).toBe('good');
    expect(getByText(/good/).tagName).toBe('MARK');
    expect(getByLabelText(/search count/).textContent).toBe('1');
  });
});
