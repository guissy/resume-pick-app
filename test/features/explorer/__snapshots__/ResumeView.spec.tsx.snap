// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeView component should match exact snapshot 1`] = `
<div
  className="wrap"
>
  <header
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    <label
      className="searchWrap"
      htmlFor="search"
    >
      <i
        className="fa fa-search"
      />
      <span
        className="count"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        0
      </span>
      <input
        className="search"
        id="search"
        onChange={[Function]}
        placeholder="输入关键字搜索简历"
        type="search"
        value=""
      />
    </label>
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "none",
          "border": "1px solid #fff",
          "color": "#fff",
          "cursor": "pointer",
          "fontSize": 14,
        }
      }
      type="button"
    >
      使用本地软件打开此文件
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-times fa-2x"
      />
    </button>
  </header>
  <article
    className="article"
  >
    <p
      style={
        Object {
          "marginBlockEnd": 5,
          "marginBlockStart": 5,
        }
      }
    >
      <span>
        <mark
          className="kw0 pos_1_0 "
          style={Object {}}
        >
          vue
        </mark>
        <span
          className=""
        >
           good
        </span>
      </span>
    </p>
  </article>
</div>
`;
